StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1648InheritDocMustBeUsedWithInheritingClass.cs

```diff
@@ -159,6 +159,56 @@ private static void HandleMemberDeclaration(SyntaxNodeAnalysisContext context)
                 }
             }
 
+            if (memberSyntax is ConstructorDeclarationSyntax constructorDeclarationSyntax)
+            {
+                ISymbol symbol = context.SemanticModel.GetDeclaredSymbol(constructorDeclarationSyntax, context.CancellationToken);
+
+                if (symbol is IMethodSymbol constructorMethodSymbol && constructorMethodSymbol.ContainingType is INamedTypeSymbol enclosingNamedTypeSymbol)
+                {
+                    INamedTypeSymbol baseType = enclosingNamedTypeSymbol.BaseType;
+
+                    if (baseType.SpecialType != SpecialType.System_Object)
```

Comment:
Do we really want to skip classes that inherits from Object? Maybe not the best documentation, but it seems to work:

![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/5803963/8085cde0-0503-4769-bed1-712601cfca6d)